code/lib/instrumenter/src/instrumenter.ts

```diff
@@ -412,12 +412,14 @@ export class Instrumenter {
     const { callRefsByResult, renderPhase } = this.getState(call.storyId);
 
     // Map complex values to a JSON-serializable representation.
-    const serializeValues = (value: any): any => {
+    // We use a depth, to avoid infinite recursion of self referencing values.
```

Comment:
This is a good improvement in general, but extra neccesary now since the story context is self-referencing.

Reply:
I suggest adding a `__context__` property like the one we have for `__class__` and `__date__`. The value could just be `null` but the deserialization logic can handle this special value.

Reply:
But that would only handle self referencing context right? In theory any object can be instrumented and be self referencing. For example when calling `await expect(object).toXXX`

Reply:
Yes, and we _could_ handle those too if we'd be checking references (e.g. by introducing a `__ref__` property with a string path as value). Would that be overkill?

Reply:
Continued this discussion in slack

Reply:
Added support for cyclical data. It would be deserialized as '[Cyclical]' but I think that is all right for the UI purposes in the addon.